### PR TITLE
VPN-5418 - Remove Websockets code

### PR DESCRIPTION
**Pull request by brizental**
The WebSockets feature was implemented as a replacement for the state polling mechanism we have in the application. 

From the [feature proposal](https://docs.google.com/document/d/1lRvgm6BiUnMtQ5YnieLmsquF5_1Ed0zaumYbKocywrc/edit?pli=1), there are two main reasons we would want to replace it:

1. Multiple requests are sent to Guardian to check if there were state changes, usually they are pointless because nothing has changed.
2. Long time for state propagation between devices ([polling in production environments is in on a 1hr timer](https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/apps/vpn/appconstants.h#L123)) putting users in weird states that are unclear and usually result in application errors.

There are multiple reasons that lead us now to want to remove this feature. 

1. Reason 1. why we wanted to implement websockets is still unadressed by websockets. Due to server configurations that are not changeable, the server resets any ongoing connection every 10 minutes. This means every connected user is re-starting the websocket connection more often than we are polling the server for changes. Therefore, the WebSockets feature has increased the number server requests made by the app, not decreased.
2. The feature itself has been shifted in priority and the epic to finalize it is in Low / Lowest priority since April/2023. The children of the epic have not been changed since January/2023. This shows a general lack of interest in the feature from product / engineering.
3. There is only one push message that was ever implemented using WebSockets: DEVICE_REMOVED. It feels like an edge case that a user would notice a one hour delay in being disconnected from a second device that they have removed and even more like an edge case that they would notice the device not disapeearing from their device list in one hour (for more data and less feelings, see point 4.). Still there are cheaper ways these scenarios can be addressed:
    1. Upon device removal we may show a message to users like "Device removal may take up to one hour to be mirrored by your other devices".
    2. When looking at devices list, we can add an "Updated at: HH:MM:SS" footer and maybe even allow users to update the device list at will.
    3. Weird error states can be addressed individually. It's known that when a device is removed and the state is not propagated, users may go into a "No Signal" state if the removed device is connected. We can implement handling for this error case and poll the server for device status once the user goes into a "No Signal" state for a given amount of time.
    4. The VPN is usually in the background, whenever an issue occurs the users are likely to foreground the app. On mobile devices whenever the application is foregrounded we poll for new state. We may want to extend this functionality to desktop so that whenever the application is opened it has the latest state.
   5. etc.

In hindsight, I believe it was a mistake in the feature proposal not to add a session listing other possible fixes for the issues proposed. We live and learn.

4. Removing this feature even without implementing any of the fallback scenarios listed in the bullet above, would still not result in a high user impact. From https://sql.telemetry.mozilla.org/dashboard/mozilla-vpn-push-messages-monitoring, notice that while almost 100% of all users do report a successfull websocket connection, less than 0.5% of users are actually receiving a push message.
5. Maintenance burden. This code has never really caused much engineering grief on the VPN client, but it still has to be dealt with by developers. It does cause some grief on the Guardian development side, especially due to the high amount of noise in server logs and errors due to the constant websocket connection being broken and restarted.

### CODE REVIEW

**Review by strseb — APPROVED** [review]
r++ delete

**Review by lesleyjanenorton — COMMENTED** [review]
For our future selves and since not everyone who might read this was present for our conversation at All Hands, could we add a summary of what we are doing here and why we want to remove this code? 

For my own clarity removing this means that:
- Device removals and account cancellations/expirations will now be caught exclusively by scheduled tasks. 
- This could mean a delay of up to four hours before a user on `device1` is signed out of the app after removing `device1` from `device2`. This is very likely already happening on mobile because the app must be foregrounded to receive the device removal message from Guardian. 

Question: Are we confident we know what happens on `device1` after getting removed by `device2`? Not a return to this, hopefully? (https://mozilla-hub.atlassian.net/browse/VPN-77)

File: tests/functional/testDevices.js [hunk 1]
```diff
@@ -290,12 +290,12 @@ describe('Devices', function() {
       await vpn.waitForQuery(
         queries.screenSettings.myDevicesView.DEVICE_LIST.visible());
 
-      const key = UserData.devices[0].pubkey;
-      UserData.devices.splice(0, 1);
-      await vpn.sendPushMessageDeviceDeleted(key);
+      // const key = UserData.devices[0].pubkey;
```
[the remark this answers is not in the document]
Comment (brizental) [reply tests/functional/testDevices.js:293]: Ha, good catch. I commented this out quickly just to bypass an error during implementation and didn't actually address it.

**Comment by brizental** [comment]
> Device removals and account cancellations/expirations will now be caught exclusively by scheduled tasks.

Correct.

> This could mean a delay of up to four hours before a user on device1 is signed out of the app after removing device1 from device2. This is very likely already happening on mobile because the app must be foregrounded to receive the device removal message from Guardian.

Yes, this currently happens on mobile and yes we would be returning to this on deskotp as well. The desktop issue can be addressed by other means such as verifying device status a few seconds after going into a "No Signal" state. I think we should add that to this PR, so we are sure we are not making user experience worse in any way. 

We already verify device status on toggling the VPN on so that should not be an issue. Something else we may consider for the case when the user does not have the VPN on is to listen for foreground events on desktop like we do on mobile and verify device state on foreground as well.

**Review by lesleyjanenorton — APPROVED** [review]
Thank you for the updated summary, Bea!
Last question: 
> The desktop issue can be addressed by other means such as verifying device status a few seconds after going into a "No Signal" state
Wouldn't this be impossible if the client is in NoSignal and the Internet is bricked? 

Regardless, IMO `Upon device removal we may show a message to users like "Device removal may take up to one hour to be mirrored by your other devices".` is enough here to mitigate any confusion experienced by those `less than 0.5%`. Since the push message is also received by anyone cancelling their account through Support in first month of their subscription to receive a refund, we should notify Support of this change and make sure they convey that it may take 'some time' for the user to be signed out of all authenticated devices.

**Comment by brizental** [comment]
>  Wouldn't this be impossible if the client is in NoSignal and the Internet is bricked?

Hm, that is a great point. I can file a ticket for us to look into this scenario more closely, WDYT?

**Comment by lesleyjanenorton** [comment]
> > Wouldn't this be impossible if the client is in NoSignal and the Internet is bricked?
> 
> Hm, that is a great point. I can file a ticket for us to look into this scenario more closely, WDYT?

Please do, thank you!

**Comment by brizental** [comment]
I have created https://mozilla-hub.atlassian.net/browse/VPN-5468 and there is already https://mozilla-hub.atlassian.net/browse/VPN-5425 which was created recently to handle the scenario where a device is removed but it is connected to the VPN.

**Pull request merged**